src/consensus/pbft_manager.cpp

```diff
@@ -969,7 +975,8 @@ void PbftManager::secondFinish_() {
   // Odd number steps from 5 are in second finish
   auto round = getPbftRound();
   LOG(log_tr_) << "PBFT second finishing state at step " << step_ << " in round " << round;
-  auto end_time_for_step = (step_ + 1) * LAMBDA_ms - POLLING_INTERVAL_ms;
+  assert(step_ >= startingStepInRound_);
+  auto end_time_for_step = (2 + step_ - startingStepInRound_) * LAMBDA_ms - POLLING_INTERVAL_ms;
```

Comment:
Same here. If step_ always equal to startingStepInRound_, end_time_for_step is always 2 * LAMBDA_ms - POLLING_INTERVAL_ms